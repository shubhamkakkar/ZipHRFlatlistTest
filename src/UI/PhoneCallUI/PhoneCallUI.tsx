import React from 'react';
import { Linking, Platform, TouchableOpacity } from 'react-native';
import styles from './styleSheetPhoneCallUI';

type TPhoneCallUI = {
  number: string;
  children: React.ReactElement;
};

export default function PhoneCallUI({ number, children }: TPhoneCallUI) {
  async function onPressPhoneCall() {
    let phoneNumber = '';
    if (Platform.OS === 'android') {
      phoneNumber = `tel:${number}`;
    } else {
      phoneNumber = `telprompt:${number}`;
    }
    await Linking.openURL(phoneNumber);
  }

  return (
    <TouchableOpacity onPress={onPressPhoneCall} style={styles.container}>
      {children}
    </TouchableOpacity>
  );
}
