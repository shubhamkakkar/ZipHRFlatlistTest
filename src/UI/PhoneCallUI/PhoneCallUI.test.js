import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';
import PhoneCallUI from './PhoneCallUI';

it('PhoneCallUI matches snapshot - props: {{ number: \'9818163250\',, }}', async () => {
  const tree = renderer.create(
    <PhoneCallUI {...{
      number: '9818163250',
    }}
    >
      <View />
    </PhoneCallUI>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
