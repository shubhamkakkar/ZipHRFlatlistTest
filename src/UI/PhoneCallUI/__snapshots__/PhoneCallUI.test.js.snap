// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneCallUI matches snapshot - props: {{ number: '9818163250',, }} 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "opacity": 1,
    }
  }
>
  <View />
</View>
`;
